clang/lib/Driver/ToolChains/BareMetal.cpp

```diff
@@ -545,9 +545,27 @@ void baremetal::Linker::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back(Arch == llvm::Triple::aarch64_be ? "-EB" : "-EL");
   }
 
-  if (!Args.hasArg(options::OPT_nostdlib, options::OPT_nostartfiles,
-                   options::OPT_r)) {
-    CmdArgs.push_back(Args.MakeArgString(TC.GetFilePath("crt0.o")));
+  bool WantCRTs =
```

Comment:
This is just a suggestion, but I think `NeedsCRT` would be more accurate.
```suggestion
  bool NeedCRT =
```

Reply:
Done